experimental/dds/tree/src/Checkout.ts

```diff
@@ -169,10 +172,37 @@ export abstract class Checkout extends EventEmitterWithErrorHandling<ICheckoutEv
 		assert(currentEdit !== undefined, 'An edit is not open.');
 		this.currentEdit = undefined;
 		const editingResult = currentEdit.close();
-		assert(editingResult.status === EditStatus.Applied, 'Locally constructed edits must be well-formed and valid');
+		if (editingResult.status !== EditStatus.Applied) {
+			const { failure } = editingResult;
+			const additionalProps: ITelemetryProperties = {};
+			switch (failure.kind) {
+				case TransactionInternal.FailureKind.BadPlace:
+					additionalProps.placeFailure = failure.placeFailure;
+					break;
+				case TransactionInternal.FailureKind.BadRange: {
+					const { rangeFailure } = failure;
+					if (typeof rangeFailure === 'string') {
+						additionalProps.rangeFailure = rangeFailure;
+					} else {
+						additionalProps.rangeFailure = rangeFailure.kind;
+						additionalProps.rangeEndpointFailure = rangeFailure.placeFailure;
+					}
+					break;
+				}
+				default:
+					break;
```

Comment:
I am wondering if there would be some value in logging the `failure.kind` in the default case. I think that with concurrency some of the other cases could arise, though it sounds like the specific user you are supporting was interested in malformed edits.

Reply:
Oh, I see it's being logged bellow, never mind. :)